### PR TITLE
ERNE _parse_txt routine

### PR DESCRIPTION
**Pull request by ankitkmr**
This function is intended to read in data from ERNE data file stored on hard disk and return the header as a list and data as pandas dataframe. I'll implement the class structure around it soon.

### CODE REVIEW

File: sunpy/lightcurve/sources/erne.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    data_modify = []
+    #Converting separate datetime element into a single datetime.datetime column
+    for i in range(len(data)):
+        if int(data['col1'][i]) >= 96:
+            year = 1900 + int(data['col1'][i])
+        else:
+            year = 2000 + int(data['col1'][i])
+       
+        data_modify = data_modify + [TimeRange(datetime(year,int(data['col2'][i]),int(data['col3'][i]),int(data['col4'][i][:2]),int(data['col4'][i][2:])), datetime(year,int(data['col2'][i]),int(data['col3'][i]),int(data['col5'][i][:2]),int(data['col5'][i][2:])))]
```
Comment (dpshelio) [inline sunpy/lightcurve/sources/erne.py]: use `data_modify.append()`

**Comment by ehsteve** [comment]
This PR requires a description. What is the purpose of this code?

**Comment by ehsteve** [comment]
Also this probably belongs to soho.py and not its own module.

File: sunpy/lightcurve/sources/erne.py (hunk 1 above)
Comment (ankitkmr) [inline sunpy/lightcurve/sources/erne.py]: Done with this, doing the same in others as well, also stored the columns in recognizable variables to use at other places.

File: sunpy/lightcurve/sources/erne.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        data_modify.append(TimeRange(date1, date2))
+    
+    #Removing the columns with separate date elements like year, day, hour, minute, second
+    data.remove_columns(['col{}'.format(i) for i in range(1,6)])
+    #Adding a single Timerange column to the data
+    data.add_column(Column(data = data_modify, name='col_1'),0)
+    
+    #To modify header
+    for i in range(len(header[5:])):
+        header[i+5] = 'Intensities [1/(cm^2*sr*s*MeV)] in energy channel ' + header[i+5] +' [MeV] '
```
Comment (dpshelio) [inline sunpy/lightcurve/sources/erne.py]: i think it's better in this way:
```python
header[5:] = ['Intensities [1/(cm^2*sr*s*MeV)] in energy channel {}  [MeV] '.format(val) for val in header[5:]]
```

File: sunpy/lightcurve/sources/erne.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    data.add_column(Column(data = data_modify, name='col_1'),0)
+    
+    #To modify header
+    for i in range(len(header[5:])):
+        header[i+5] = 'Intensities [1/(cm^2*sr*s*MeV)] in energy channel ' + header[i+5] +' [MeV] ' 
+    header = ['TimeRange'] + header[5:]
+    
+    # To add the column names in the astropy table object
+    for key2 in range(len(data.colnames)):
+        data.rename_column(data.colnames[key2], header[key2])
```
Comment (dpshelio) [inline sunpy/lightcurve/sources/erne.py]: I don't think it's to pythonic to have this constructions like `range(len(xx))`
wouldn't be better to do somtething like:
```python
for elem, head_key in enumerate(header):
    data.rename_column(data.colnames[elem], head_key)
```

File: sunpy/lightcurve/sources/erne.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    http://srl.utu.fi/erne_data/carrot/carrotp.html
+
+    """
+    
+    #Reading in Data along with header
+    data = ascii.read(filepath, delimiter = "\s", data_start = 2 ) 
+
+    header = data.colnames
+    for i in range(len(header)):
+        header[i] = data[0][header[i]]
```
Comment (dpshelio) [inline sunpy/lightcurve/sources/erne.py]: I don't like this either...  - though I don't understand what's going on anyway... I think you could do:
```python
header = [data[0][key] for key in data.colnames]
```

Comment (ankitkmr) [inline sunpy/lightcurve/sources/erne.py]: made all the changes :)

**Pull request closed without merging**